font/src/ft/mod.rs

```diff
@@ -529,6 +589,86 @@ impl FreeTypeRasterizer {
     }
 }
 
+fn downsample_bitmap(mut bitmap_glyph: RasterizedGlyph, fixup_factor: f64) -> RasterizedGlyph {
+    // Don't try to upscale
+    if fixup_factor > 1.0 {
+        return bitmap_glyph;
+    }
+
+    let bitmap_width = bitmap_glyph.width as f64;
+    let bitmap_height = bitmap_glyph.height as f64;
+
+    let width = (bitmap_width * fixup_factor) as usize;
+    let height = (bitmap_height * fixup_factor) as usize;
+
+    let bitmap_width = bitmap_width as usize;
+    let bitmap_height = bitmap_height as usize;
+
+    let b_buf = &bitmap_glyph.buf;
+    let scaling_factor =
+        (bitmap_width as f32 / width as f32).max(bitmap_height as f32 / height as f32);
+    let advance_step = scaling_factor.ceil() as usize;
+    let mut scaled_buffer = Vec::with_capacity(width * height * 3);
+
+    let mut new_line_index = 0;
+    let mut source_line_index = 0;
+
+    while new_line_index < height {
+        let mut new_column_index = 0;
+        let mut source_column_index = 0;
+
+        while new_column_index < width {
+            let mut r: u32 = 0;
+            let mut g: u32 = 0;
+            let mut b: u32 = 0;
+            let mut pixels_picked: u32 = 0;
+
+            let source_end_line = std::cmp::min(source_line_index + advance_step, bitmap_height);
+            let source_end_column = std::cmp::min(source_column_index + advance_step, bitmap_width);
```

Comment:
Usually it's fine to import even `std::cmp::min` at the top, that makes these super simple functions a little easier. `std::cmp` would be fine too, so we'd use `cmp::min`, but since it's used so often in some modules the `min` function is basically a reserved name anyways, so it shouldn't be confusing to anyone.